File: 卡码/代码随想录/Python/动态规划/04-股票问题/LeetCode_0122.py
#122. 买卖股票的最佳时机 II
from typing import *

class Solution:
  def maxProfit(self, prices: List[int]) -> int:
    dp:List[int]=[-prices[0],0]
    for i in range(1,len(prices)):
      dp[0]=max(dp[0],dp[1]-prices[i])
      dp[1]=max(dp[1],dp[0]+prices[i])
    return dp[1]

File: 卡码/代码随想录/JS/动态规划/04-股票问题/LeetCode_0122.js
//122. 买卖股票的最佳时机 II
const maxProfit = function(prices) {
  const dp=[-prices[0],0];
	for(let i=1;i<prices.length;i++){
		dp[0]=Math.max(dp[0],dp[1]-prices[i]);
		dp[1]=Math.max(dp[1],dp[0]+prices[i]);
	}
	return dp[1];
};
